components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthenticatorAdapterService.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework;
+
+import org.wso2.carbon.identity.application.common.model.FederatedAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.LocalAuthenticatorConfig;
+
+/**
+ * Authenticator Adapter Service which responsible for creating ApplicationAuthenticator for the provided user defined
+ * authenticator configs.
+ */
+public interface AuthenticatorAdapterService {
+
+    /**
+     * Get the ApplicationAuthenticator for the given user defined federated authenticator config.
+     *
+     * @param config    Federated Authenticator Config.
+     * @return  FederatedApplicationAuthenticator instance.
+     */
+    FederatedApplicationAuthenticator getFederatedAuthenticatorAdapter(FederatedAuthenticatorConfig config);
```

Comment:
Shall we rename to getUserDefinedFederatedAuthenticator and pass the UserdefinedFederatedAuthConfig as input